### PR TITLE
Shadowlamb: fix wrong item amount in mounts after pop

### PR DESCRIPTION
**Pull request by tehron**
A fix in best Shadowlamb code quality

### CODE REVIEW

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/cmd/mount/mount.php [hunk 1]
```diff
@@ -257,7 +257,7 @@ private static function on_pop(SR_Player $player, array $args)
 				return false;
 			}
 
-			if (false === $item->useAmount($player, $amt))
+			if (false === $item->useAmount($player, $amt, true, true))
```
Comment (livinskull) [inline core/module/Dog/dog_modules/dog_module/Shadowlamb/cmd/mount/mount.php:260]: I don't think you'd want to have `modify = true` here. That should happen only on `#use`, if I understand this stuff correctly.

Comment (tehron) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/cmd/mount/mount.php:260]: True preserves the current behavior.

Comment (dloser) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/cmd/mount/mount.php:260]: Hmm, I don't think `modify()` does anything with mount, does it?

Also, `modify()` is called at the end of this function (line 299), after the `giveItem()` call, so having it called here has no real use, I'd say.

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php [hunk 2]
```diff
@@ -2081,9 +2081,16 @@ public function giveItem(SR_Item $item)
...
+	public function itemAmountChanged(SR_Item $item, $amount_change, $modify=true, $use_mount=false)
+	{
+        if ($use_mount)
+        {
+            $this->sr4_mount_inv->itemAmountChanged($item, $amount_change);
+        }
+        else
+        {
+            $this->sr4_inventory->itemAmountChanged($item, $amount_change);
+        }
```
Comment (dloser) [inline core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php:2093]: Mixing tabs and spaces.

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php [hunk 3]
```diff
@@ -2081,9 +2081,16 @@ public function giveItem(SR_Item $item)
 		return $this->sr4_inventory->addItem($item);
 	}
 
-	public function itemAmountChanged(SR_Item $item, $amount_change, $modify=true)
-	{
-		$this->sr4_inventory->itemAmountChanged($item, $amount_change);
+	public function itemAmountChanged(SR_Item $item, $amount_change, $modify=true, $use_mount=false)
```
Comment (dloser) [inline core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php:2084]: If `modify()` is indeed not needed for changes to mount, then it is probably nicer to have a separate `mountItemAmountChanged()` or something.

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Item.php [hunk 4]
```diff
@@ -532,7 +532,7 @@ private function displayWorth()
 	 * @param int $amount
 	 * @return true|false
 	 */
-	public function useAmount(SR_Player $player, $amount=1, $modify=true)
+	public function useAmount(SR_Player $player, $amount=1, $modify=true, $use_mount=false)
```
Comment (dloser) [inline core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Item.php:535]: `$use_mount` -> `$item_from_mount' or something, perhaps?

Not very nice, but not sure how to improve other than duplicating this code. :/

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php (hunk 3 above)
Comment (livinskull) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php:2084]: A separate method is imho always preferable to a boolean argument switching the implementation.
Especially if the method only switches on that boolean argument.

But then again, this code is a mess already :)

Comment (tehron) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php:2084]: "a mess", that's why I didn't bother to nicen things up for a fix.We all know the code is unmaintainable.
And we all should know default parameters are bad and bite you sooner or later, but it's already in use, so I added another default parameter.

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php (hunk 2 above)
Comment (tehron) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php:2093]: Hard to adopt gizmore's code style.
I'll fix...

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/cmd/mount/mount.php (hunk 1 above)
Comment (tehron) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/cmd/mount/mount.php:260]: You're right, @dloser, I'll consider calling `useAmount()` with `$modify=false`

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Inventory.php [hunk 5]
```diff
@@ -775,4 +775,8 @@ public function swapItems($name1, $name2, $requesting_player)
 		$this->onChanged(true);
 	}
 
+	public function contains($item)
```
Comment (dloser) [inline core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Inventory.php:778]: Figured that would be to expensive to implement. Much nicer now!

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php [hunk 6]
```diff
@@ -2083,7 +2083,19 @@ public function giveItem(SR_Item $item)
...
+		{
+			$this->sr4_inventory->itemAmountChanged($item, $amount_change);
+		}
+		elseif ($this->sr4_mount_inv->contains($item))
+		{
+			$this->sr4_mount_inv->itemAmountChanged($item, $amount_change);
+		}
+		else
+		{
+			// TODO gizmore print error message (tehron)
```
Comment (dloser) [inline core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php:2096]: `GWF_Mail::sendMailS(GWF_BOT_EMAIL, 'gizmore@wechall.net', '[Shadowlamb] TODO gizmore print error message (tehron)', '-nt-');`

:stuck_out_tongue:

Comment (dloser) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php:2096]: Anyway, should not happen, so I'm not against throwing an exception here. Only way anyone is going to notice, probably. (And prevents users from using it as an infinite source of items.)

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php (hunk 3 above)
Comment (tehron) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php]: resolved with new commit

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Item.php (hunk 4 above)
Comment (tehron) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Item.php]: resolved with new commit

File: core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php (hunk 6 above)
Comment (tehron) [reply core/module/Dog/dog_modules/dog_module/Shadowlamb/core/SR_Player.php]: resolved in 551da4d

**Pull request merged**